media/js/src/solutionStep.tsx

```diff
@@ -59,38 +63,107 @@ export const SolutionStep: React.FC<SolutionStepProps> = (
 
     const handleStatementInput = (
         evt: React.ChangeEvent<HTMLInputElement>): void => {
-        setStatementInput(evt.target.value);
+        setNextStep(evt.currentTarget.value);
         setError('');
     };
 
     const handleLawSelect = (
         evt: React.ChangeEvent<HTMLSelectElement>): void => {
-        setSelectedLaw(evt.target.value);
+        setNextRule(evt.target.value);
+    };
+
+    async function validateStep() {
+        const hintData: HintData = {
+            next_expr: '',
+            rule: '',
+            step_list: [''],
+            answer: ''
+        };
+        let lastCorrectStep = '';
+
+        if(stepList.length > 0) {
+            lastCorrectStep = stepList[stepList.length - 1][1];
+        } else {
+            lastCorrectStep = statement.question;
+        }
+
+        hintData['next_expr'] = latex2raw(nextStep);
+        hintData['rule'] = nextRule.toLocaleLowerCase();
+        hintData['step_list'] =[latex2raw(lastCorrectStep)];
+        hintData['answer'] = statement.answer;
+        // eslint-disable-next-line max-len
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const toolsData: Tools = await getHints(hintData);
+
+        void validation(toolsData);
+    }
+
+    const validation = (respData: Tools) =>{
+
+        if(!respData.isValid) {
+            setError(respData.errorMsg);
+        } else if(respData.isValid && !respData.isSolution) {
+
+            //If the input is valid and not the solution, add to stepList
+            //Change status to in progress.
+            const step: [string, string] = [nextRule, nextStep];
+            const newStepList: [string, string][] = updateLocalStepList(
+                id, step);
+            updateLocalQuestionStatus(id, 'inprogress');
+            setStepList(newStepList);
+            setHint(['', '']);
+            setHintButtonCount(0);
+
+        } else if(respData.isValid && respData.isSolution) {
+
+            //If the input is valid add to stepList,
+            //and if completed change status to complete
+            const step: [string, string] = [nextRule, nextStep];
+            const newStepList = updateLocalStepList(id, step);
+            updateLocalQuestionStatus(id, 'complete');
+            setStepList(newStepList);
+            setHint(['', '']);
+            setHintButtonCount(2);
+        }
     };
 
     const handleSubmit = (
         evt: React.MouseEvent<HTMLButtonElement>
     ): void => {
         evt.preventDefault();
-        if(!statementInput) {
+        if(hintButtonCount === 2) {
+            setHintButtonCount(0);
+        }
+        if(!nextStep) {
             setError('Please enter a statement.');
+        } else {
+            setError('');
+            void validateStep();
```

Comment:
Is the `void` needed here?

Reply:
I believe void is best here since that async function isn't returning anything.